### PR TITLE
[YUNIKORN-1371] Migrate to new ConfigMap configuration

### PR DESCRIPTION
**Pull request by craigcondit**
Updates Helm charts to deploy new yunikorn-defaults ConfigMap, and removes legacy configuration environment variables.
This also fixes an issue with RBAC only being deployed on install and not upgrade.

JIRA: https://issues.apache.org/jira/browse/YUNIKORN-1371

### CODE REVIEW

**Review by pbacsko — APPROVED** [review]
+1 LGTM

File: helm-charts/yunikorn/templates/deployment.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 spec:
-  replicas: {{ .Values.replicaCount }}
+  replicas: 1
```
Comment (elihschiff) [inline helm-charts/yunikorn/templates/deployment.yaml:27]: Hi @craigcondit 

I am just curious about the decision on why the replicas value is now hard coded. Does this feature no longer work with values > 1? Thanks!

Comment (craigcondit) [reply helm-charts/yunikorn/templates/deployment.yaml:27]: @elihschiff, it never did (for the scheduler itself), only for the admission controller, which has its own variable. Setting it >1 causes a lot of problems, so it was removed. YuniKorn is designed to recover very rapidly without any shared state, so having two active schedulers is not necessary.

Comment (elihschiff) [reply helm-charts/yunikorn/templates/deployment.yaml:27]: Got it, thanks for the quick reply!

**Pull request closed without merging**